### ISSUE TITLE
Problem with docker api breaks web UI

### ISSUE THREAD

**Issue by Ringelnatz** [mailcow/mailcow-dockerized#981]
Since I upgraded to the latest docker version 17.12 I experience some strange problems with docker. Some containers become unable to manage after some time, docker inspect/attach/logs/stop etc. do not work anymore, docker-compose runs in a timeout and a restart of the host is required. This is really annoying and I did not find any solution yet.

This problem causes some errors with the `dockerapi-mailcow` container:
```
dockerapi-mailcow_1  | [2018-01-29 14:39:27,953] ERROR in app: Exception on /containers/json [GET]
dockerapi-mailcow_1  | Traceback (most recent call last):
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1612, in full_dispatch_request
dockerapi-mailcow_1  |     rv = self.dispatch_request()
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1598, in dispatch_request
dockerapi-mailcow_1  |     return self.view_functions[rule.endpoint](**req.view_args)
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/site-packages/flask_restful/__init__.py", line 480, in wrapper
dockerapi-mailcow_1  |     resp = resource(*args, **kwargs)
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/site-packages/flask/views.py", line 84, in view
dockerapi-mailcow_1  |     return self.dispatch_request(*args, **kwargs)
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/site-packages/flask_restful/__init__.py", line 595, in dispatch_request
dockerapi-mailcow_1  |     resp = meth(*args, **kwargs)
dockerapi-mailcow_1  |   File "/server.py", line 25, in get
dockerapi-mailcow_1  |     return jsonify(type='danger', msg=e)
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/site-packages/flask/json.py", line 263, in jsonify
dockerapi-mailcow_1  |     (dumps(data, indent=indent, separators=separators), '\n'),
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/site-packages/flask/json.py", line 123, in dumps
dockerapi-mailcow_1  |     rv = _json.dumps(obj, **kwargs)
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/json/__init__.py", line 251, in dumps
dockerapi-mailcow_1  |     sort_keys=sort_keys, **kw).encode(obj)
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/json/encoder.py", line 209, in encode
dockerapi-mailcow_1  |     chunks = list(chunks)
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/json/encoder.py", line 434, in _iterencode
dockerapi-mailcow_1  |     for chunk in _iterencode_dict(o, _current_indent_level):
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
dockerapi-mailcow_1  |     for chunk in chunks:
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/json/encoder.py", line 442, in _iterencode
dockerapi-mailcow_1  |     o = _default(o)
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/site-packages/flask/json.py", line 80, in default
dockerapi-mailcow_1  |     return _json.JSONEncoder.default(self, o)
dockerapi-mailcow_1  |   File "/usr/local/lib/python2.7/json/encoder.py", line 184, in default
dockerapi-mailcow_1  |     raise TypeError(repr(o) + " is not JSON serializable")
dockerapi-mailcow_1  | TypeError: ReadTimeout(ReadTimeoutError("UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)",),) is not JSON serializable
dockerapi-mailcow_1  | 46.38.xx.xx - - [29/Jan/2018 14:39:27] "GET /containers/json HTTP/1.1" 500 -
```
Related to this: When I open `debug.php` the web UI gets completely unresponsive until I restart the `php-fpm-mailcow` container (or wait for some minutes).

I know this is really edge case, but maybe the api client could be improved to handle such docker failures.

**Comment by andryyy** [comment]
It should have a proper timeout, yes. 

I cannot reproduce that issue. What's in your syslog at the time it happens? 

Did you stop and remove all containers before upgrading Docker? If not, run `docker-compose down`, restart the daemon and `up -d` the stack.

**Comment by Ringelnatz** [comment]
I don't think that my issues are reproducible. There are very few information on this (I found https://github.com/moby/moby/issues/35933 and https://github.com/docker/for-mac/issues/2378 (although I am not using Mac)). I did not stop or remove any containers before docker upgrade (should I?), but did this several times since that time. Since the upgrade my syslog is full of `level=warning msg="unknown container"` messages, no idea if this is related or not.

Meanwhile I migrated my mailcow to another host and will reinstall the "broken" host soon. This issue only aims at improving exception handling to avoid errors like the above and an unresponsive web interface.
